File: problem_solving/easy/breaking_records/breaking_records.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the breakingRecords function below.
def breakingRecords(scores)
    first = [scores[0], scores[0]]
    scores.each_with_object({max: 0, min: 0}) do |score, hash|
        p "hash: #{hash} score: #{score} first: #{first}"
        if score < first[1]
            hash[:min] += 1
            first[1] = score
        end
        if score > first[0]
            hash[:max] += 1
            first[0] = score
        end
    end.values
end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

n = gets.to_i

scores = gets.rstrip.split(' ').map(&:to_i)

result = breakingRecords scores

fptr.write result.join " "
fptr.write "\n"

fptr.close()

File: problem_solving/easy/breaking_records/breaking_records.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the breakingRecords function below.
function breakingRecords(scores) {
    let maxMin = [0, 0];
    let first = [scores[0], scores[0]];
    for(let i = 0; i < scores.length; i++){
        if (scores[i] > first[0]){
            maxMin[0]++;
            first[0] = scores[i];
        }
        if (scores[i] < first[1]){
            maxMin[1]++;
            first[1] = scores[i];
        }
    }
    return maxMin;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    const scores = readLine().split(' ').map(scoresTemp => parseInt(scoresTemp, 10));

    const result = breakingRecords(scores);

    ws.write(result.join(' ') + '\n');

    ws.end();
}
